apps/trivium/benches/casting.rs

```diff
@@ -0,0 +1,81 @@
+use tfhe::shortint::prelude::*;
+
+use rayon::prelude::*;
+
+use criterion::{criterion_group, criterion_main, Criterion};
+
+pub fn pack_cast_64(c: &mut Criterion) {
+    let (client_key_1, server_key_1): (ClientKey, ServerKey) = gen_keys(PARAM_MESSAGE_1_CARRY_1);
+    let (client_key_2, server_key_2): (ClientKey, ServerKey) = gen_keys(PARAM_MESSAGE_2_CARRY_2);
+
+    let ksk = KeySwitchingKey::new(
+        (&client_key_1, &server_key_1),
+        (&client_key_2, &server_key_2),
+        PARAM_KEYSWITCH_1_1_TO_2_2,
+    );
+
+    let vec_ct = vec![client_key_1.encrypt(1); 64];
+
+    c.bench_function("pack_cast_64", |b| {
+        b.iter(|| {
+            let _ = (0..32)
+                .into_par_iter()
+                .map(|i| {
+                    let byte_idx = 7 - i / 4;
+                    let pair_idx = i % 4;
+
+                    let b0 = &vec_ct[8 * byte_idx + 2 * pair_idx];
+                    let b1 = &vec_ct[8 * byte_idx + 2 * pair_idx + 1];
+
+                    ksk.cast(
+                        &server_key_1.unchecked_add(b0, &server_key_1.unchecked_scalar_mul(b1, 2)),
+                    )
+                })
+                .collect::<Vec<_>>();
```

Comment:
Feels strange to redefine the function in the bench, when its already in transcipher_from_1_1_stream

Reply:
It's close, but here you don't do the Fheuint64::xor
I thought about factorizing this part in a function, but it would make the main code a little weirder, to only improve a bit the bench code, it didn't seem worth it